manipulation/src/torch/csrc/jit/import.cpp

```diff
@@ -0,0 +1,228 @@
+#include "torch/csrc/jit/import.h"
+
+namespace torch { namespace jit {
+
+std::unique_ptr<Graph> graphProtoToGraph(const onnx::GraphProto& gp);
+
+at::ScalarType protoTypeToATenType(onnx::TensorProto_DataType t) {
+  switch(t) {
+  case onnx::TensorProto_DataType_FLOAT:
+    return at::kFloat;
+  case onnx::TensorProto_DataType_INT8:
+    return at::kByte;
+  case onnx::TensorProto_DataType_INT16:
+    return at::kShort;
+  case onnx::TensorProto_DataType_INT32:
+    return at::kInt;
+  case onnx::TensorProto_DataType_INT64:
+    return at::kLong;
+  case onnx::TensorProto_DataType_FLOAT16:
+    return at::kHalf;
+  case onnx::TensorProto_DataType_DOUBLE:
+    return at::kDouble;
+  case onnx::TensorProto_DataType_UNDEFINED:
+  case onnx::TensorProto_DataType_UINT8:
+  case onnx::TensorProto_DataType_UINT16:
+  case onnx::TensorProto_DataType_STRING:
+  case onnx::TensorProto_DataType_BOOL:
+  case onnx::TensorProto_DataType_UINT32:
+  case onnx::TensorProto_DataType_UINT64:
+  case onnx::TensorProto_DataType_COMPLEX64:
+  case onnx::TensorProto_DataType_COMPLEX128:
+    abort();
+  }
+}
+
+std::pair<at::Tensor, std::string> tensorProtoToTensor(const onnx::TensorProto & tp) {
+  at::ScalarType type = protoTypeToATenType(tp.data_type());
+
+  std::vector<int64_t> dims;
```

Comment:
Pre-allocate space with reserve() since you know this is going to end up being `tp.dims_size()` elements long

Reply:
will add all the reserve() calls